internal/observability/call.go

```diff
@@ -91,24 +91,48 @@ func (c call) endStats(elapsed time.Duration, err error, isApplicationError bool
 		}
 		return
 	}
-	// Bad request errors are the caller's fault.
-	if yarpcerrors.IsInvalidArgument(err) {
+
+	errCode := yarpcerrors.ErrorCode(err)
+	switch errCode {
+	case yarpcerrors.CodeInvalidArgument,
+		yarpcerrors.CodeNotFound,
+		yarpcerrors.CodeAlreadyExists,
+		yarpcerrors.CodePermissionDenied,
+		yarpcerrors.CodeOutOfRange,
+		yarpcerrors.CodeUnauthenticated:
 		c.edge.callerErrLatencies.Observe(elapsed)
```

Comment:
we can probably have a very heated argument whether these should be "caller" or "server" errors, and at the end of the day, it won't be perfect...